### ISSUE TITLE
Adding null extensions and contained shouldn't crash the serializer

### ISSUE THREAD

**Issue by wdebeau1** [ewoutkramer/fhir-net-api#38]
As discussed at the connectathon, This test should pass:
        [TestMethod]
        public void TestNullExtensionRemoval()
        {
            var p = new Patient
            {
                Extension = new List<Extension>
                {
                    new Extension("http://hl7.org/fhir/Profile/iso-21090#qualifier", new Code("VV")),
                    null
                }
            };

```
        var xml = FhirSerializer.SerializeResourceToXml(p);
    }
```

This will currently throw from DispatchingWriter:49

**Comment by brianpos** [comment]
Would it be preferable to change the Extension constructor here throw the exception (so that doesn't get to the other serializer)
I would like to do this for most of the datatypes, throwing an ArgumentException for the fields with invalid values. (including dates/times/ids too)

**Comment by ewoutkramer** [comment]
Yeah, the constructor should throw an ArgumentNullException because this is structurally invalid.

Not too sure about invalid values, since this would make it hard to parse (and correct) a file using the POCO's. Alternatively (but just later this year), you could use the DOM-based FHIR reader (which is more flexible in accepting illegal stuff, since it does not use POCO's) to do this, and make the POCO's stricter....

**Comment by ewoutkramer** [comment]
Will take this up as part of doing #356

**Comment by ewoutkramer** [comment]
Fixed in a811a9f65ff98a9024fd2371428f82463cd0edf5
